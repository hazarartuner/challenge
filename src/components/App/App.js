import React from "react";

export default () => {
  return <div className="app-component">App Component</div>;
};
