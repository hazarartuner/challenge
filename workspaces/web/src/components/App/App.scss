body {
  background: #222;
  color: #fff;
  font-size: 62.5%;
}

.app__component {
  font-size: 1.2rem;
}
