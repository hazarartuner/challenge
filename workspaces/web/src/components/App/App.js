import React from 'react';
import './App.scss';

export default () => {
  return <div className="app__component">App Component</div>;
};
