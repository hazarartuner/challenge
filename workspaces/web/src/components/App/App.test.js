import React from 'react';
import { mount } from 'enzyme';
import App from './App';
import AppIndirect from './index';

describe('Components -> App', () => {
  it('Direct and Indirect imports should be same', () => {
    expect.assertions(1);

    expect(AppIndirect).toBe(App);
  });

  it('Should match with the snapshot', () => {
    // Cant use shallow rendering technique because of the react hooks:
    // https://github.com/airbnb/enzyme/issues/2011
    const app = mount(<App />);

    expect(app).toMatchSnapshot();
  });
});
