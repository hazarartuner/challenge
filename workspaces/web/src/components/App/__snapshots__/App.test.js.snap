// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> App Should match with the snapshot 1`] = `
<_default>
  <div
    className="app__component"
  >
    App Component
  </div>
</_default>
`;
